### ISSUE TITLE
Green's function normalisation constant in wrong place

### ISSUE THREAD

**Issue by ehkropf** [ehkropf/SKPrime#42]
It appears `greensC0` (and `greensCj`) are having the normalisation constant applied only when evaluating the full function, but not the "hat" function. The normalisation should be applied to the hat function always, since this is the BVP solution.

TL;DR: The "hat" function gives the wrong values.

**Comment by ehkropf** [comment]
This fix breaks `skprime` as evidenced by running `skptests primeBasic/*`. It appears this is on points outside unit domain.

Note the commit where the fix lives is not on the master branch. See branch [green_normalise](/ehkropf/SKPrime/tree/green_normalise).

**Comment by ehkropf** [comment]
Outer point problem confirmed:

``` matlab
dv = [
  -0.2517+0.3129i
   0.2307-0.4667i];
qv = [
  0.2377
  0.1557];
alpha = -0.31837-0.44082i;

w = skprime(alpha, dv, qv);

% Check against product formula.
L = 6;
wp = skprod(dv, qv, L);
wphat = @(z) wp(z, alpha)./(z - alpha);

zp = 0.4793+0.4828i;
zp = [zp; 1/conj(zp)];

err = wphat(zp) - w.hat(zp);
```

where

```
err =

      -1.86885043329532e-06 -  2.52521160475083e-08i
         0.0679702753652227 -      0.47787048749888i
```

The BVP for `1/conj(alpha)` is thus not being formed correctly.

**Comment by ehkropf** [comment]
Note replacing the parameter in the above example with `1/conj(-0.31837-0.44082i)` reinforces this.

**Comment by ehkropf** [comment]
Fixed with merge commit 903bd65.
